client/src/ExtensionState.ts

```diff
@@ -215,12 +248,8 @@ export class State {
     public static dispose(): Promise<any> {
         try {
             return new Promise((resolve, reject) => {
-                Log.log("Ask language server to shut down.", LogLevel.Info);
-                State.client.sendRequest(Commands.Dispose, null).then(() => {
-                    Log.log("Language server has shut down, terminate the connection", LogLevel.Info);
-                    this.languageServerDisposable.dispose();
-                    resolve();
-                });
+                Log.log("Initiating language server shutdown.", LogLevel.Info);
+                State.client.stop()

```

Comment:
Why is it not necessary anymore to shut down the server?

Reply:
This is implicitly / automatically taken care of by taken care of by the LSP. The call to stop() initiates a "termination handshake" between client and server. The server is shut down during the course of that handshake.